vos/vos/commands/vsync.py

```diff
@@ -37,13 +40,265 @@
+    :return: TransferReport()
+    """
+    result = TransferReport()
+    src_md5 = None
+    stat = os.stat(src)
+    if not opt.ignore_checksum and not opt.overwrite:
+        src_md5 = compute_md5(src)
+    client = get_client(opt.certfile, opt.token)
+    if not opt.overwrite:
+        # Check if the file is the same
+        try:
+            node_info = None
+            if opt.cache_nodes:
+                node_info = global_md5_cache.get(dest)
+            if node_info is None:
+                logging.debug('Getting node info from VOSpace')
+                logging.debug(str(node_dict.keys()))
+                logging.debug(str(dest))
+                node = client.get_node(dest, limit=None)
+                dest_md5 = node.props.get(
+                    'MD5', 'd41d8cd98f00b204e9800998ecf8427e')
+                dest_length = node.attr['st_size']
+                dest_time = node.attr['st_ctime']
+                if opt.cache_nodes:
+                    global_md5_cache.update(
+                        dest,
+                        dest_md5,
+                        dest_length,
+                        dest_time)
+            else:
+                dest_md5 = node_info[0]
+                dest_length = node_info[1]
+                dest_time = node_info[2]
+            logging.debug('Destination MD5: {}'.format(
+                dest_md5))
+            if ((not opt.ignore_checksum and src_md5 == dest_md5) or
+                    (opt.ignore_checksum and
+                     dest_time >= stat.st_mtime and
+                     dest_length == stat.st_size)):
+                logging.info('skipping: {}  matches {}'.format(src, dest))
+                result.files_skipped = 1
+                result.bytes_skipped = dest_length
+                return result
+        except (transfer_exceptions.AlreadyExistsException,
+                transfer_exceptions.NotFoundException):
+            pass
+    logging.info('{} -> {}'.format(src, dest))
+    try:
+        client.copy(src, dest, send_md5=True)
+        node = client.get_node(dest, limit=None)
+        dest_md5 = node.props.get(
+            'MD5', 'd41d8cd98f00b204e9800998ecf8427e')
+        dest_length = node.attr['st_size']
+        dest_time = node.attr['st_ctime']
+        if opt.cache_nodes:
+            global_md5_cache.update(dest,
+                                    dest_md5,
+                                    dest_length,
+                                    dest_time)
+        result.files_sent += 1
+        result.bytes_sent += stat.st_size
+        return result
+    except (IOError, OSError) as exc:
+        logging.error(
+            'Error writing {} to server, skipping'.format(src))
+        logging.error(str(exc))
+        if re.search('NodeLocked', str(exc)) is not None:
+            logging.error(
+                ('Use vlock to unlock the node before syncing '
+                 'to {}').format(dest))
+    result.files_erred += 1
```

Comment:
Either way would be fine. Right now success path ends at l171 - everything after is an error. It would make a difference if we were to add another `except` in which case is it's better to have the error report at the end of the method and not repeated in each `except`.